tests/phpunit/tests/rest-api/wpRestTemplateRevisionsController.php

```diff
@@ -0,0 +1,440 @@
+	 * Admin user ID.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var int
+	 */
+	private static $admin_id;
+
+	/**
+	 * Contributor user ID.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var int
+	 */
+	private static $contributor_id;
+
+	/**
+	 * Template post.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var WP_Post
+	 */
+	private static $template_post;
+
+	/**
+	 * @var array
+	 */
+	private static $revisions = array();
+
+	/**
+	 * Create fake data before our tests run.
+	 *
+	 * @param WP_UnitTest_Factory $factory Helper that lets us create fake data.
+	 */
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+
+		self::$admin_id = $factory->user->create(
+			array(
+				'role' => 'administrator',
+			)
+		);
+		wp_set_current_user( self::$admin_id );
+
+		self::$contributor_id = $factory->user->create(
+			array(
+				'role' => 'contributor',
+			)
+		);
+
+		// Set up template post.
+		self::$template_post = $factory->post->create_and_get(
+			array(
+				'post_type'    => self::PARENT_POST_TYPE,
+				'post_name'    => self::TEMPLATE_NAME,
+				'post_title'   => 'My Template',
+				'post_content' => 'Content',
+				'post_excerpt' => 'Description of my template',
+				'tax_input'    => array(
+					'wp_theme' => array(
+						self::TEST_THEME,
+					),
+				),
+			)
+		);
+		wp_set_post_terms( self::$template_post->ID, self::TEST_THEME, 'wp_theme' );
+
+		// Update post to create a new revisions.
+		self::$revisions[] = _wp_put_post_revision(
+			array(
+				'ID'           => self::$template_post->ID,
+				'post_content' => 'Content revision #2',
+			)
+		);
+
+		// Update post to create a new revisions.
+		self::$revisions[] = _wp_put_post_revision(
+			array(
+				'ID'           => self::$template_post->ID,
+				'post_content' => 'Content revision #3',
+			)
+		);
+
+		// Update post to create a new revisions.
+		self::$revisions[] = _wp_put_post_revision(
+			array(
+				'ID'           => self::$template_post->ID,
+				'post_content' => 'Content revision #4',
+			)
+		);
+
+		// Update post to create a new revisions.
+		self::$revisions[] = _wp_put_post_revision(
+			array(
+				'ID'           => self::$template_post->ID,
+				'post_content' => 'Content revision #5',
+			)
+		);
+	}
+
+	public static function wpTearDownAfterClass() {
+		foreach ( self::$revisions as $revision ) {
+			// Also deletes revisions.
+			wp_delete_post( $revision, true );
+		}
+	}
+
+	/**
+	 * @covers WP_REST_Template_Revisions_Controller::register_routes
+	 * @ticket 56922
+	 */
+	public function test_register_routes() {
+		$routes = rest_get_server()->get_routes();
+		$this->assertArrayHasKey(
+			'/wp/v2/templates/(?P<parent>([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)[\/\w%-]+)/revisions',
+			$routes,
+			'Revisions route does not exist.'
+		);
+		$this->assertArrayHasKey(
+			'/wp/v2/templates/(?P<parent>([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)[\/\w%-]+)/revisions/(?P<id>[\d]+)',
+			$routes,
+			'Single revision based on the given ID route does not exist.'
+		);
```

Comment:
```suggestion
		$this->assertArrayHasKey(
			'/wp/v2/templates/(?P<parent>([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)[\/\w%-]+)/revisions',
			$routes,
			'Template revisions route does not exist.'
		);
		$this->assertArrayHasKey(
			'/wp/v2/templates/(?P<parent>([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)[\/\w%-]+)/revisions/(?P<id>[\d]+)',
			$routes,
			'Single template revision based on the given ID route does not exist.'
		);
		$this->assertArrayHasKey(
			'/wp/v2/template-parts/(?P<parent>([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)[\/\w%-]+)/revisions',
			$routes,
			'Template part revisions route does not exist.'
		);
		$this->assertArrayHasKey(
			'/wp/v2/template-parts/(?P<parent>([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)[\/\w%-]+)/revisions/(?P<id>[\d]+)',
			$routes,
			'Single template part revision based on the given ID route does not exist.'
		);
```

@anton-vlasenko this is what I had in mind.

Reply:
@spacedmonkey I thought the template parts related tests would go into a separate test class. As far as I recall, we discussed that they could be added later. However, I don't mind adding them here. I'll commit the changes you proposed.